src/common/nlmeans_core.c

```diff
@@ -405,7 +405,7 @@ void nlmeans_denoise(const float *const inbuf, float *const outbuf,
 #endif /* CACHE_PIXDIFFS */
   const int padded_scratch_size = 16*((scratch_size+15)/16); // round up to a full cache line
   const int numthreads = dt_get_num_threads() ;
```

Comment:
size_t

Reply:
sure... @ralfbrown - I need your recommendation on this, I think @dterrahe is right here.

Reply:
Sure `size_t numthreads`.  If you want to be thorough, `scratch_size` and `padded_scratch_size` could also be made `size_t`.  There's a second copy of that code for the SSE path near line 630.

Reply:
OK, I'll work on that tomorrow then! Thank you @dterrahe & @ralfbrown !